File: Frontend/src/app/models/computer.ts
import { Computer_ComputerPart } from "./computer_ComputerPart";
import { Item } from "./item";

/*
*This is the object structure of the computer object.
*It is used to store it in the database from frontend to backend.
*/

export interface Computer {
    id: number, 
    // *get the item object from the item.ts file
    item?: Item,
    // *get the computer_ComputerPart object from the computer_ComputerPart.ts file
    computer_ComputerParts?: Computer_ComputerPart[]
}

File: Backend/SOP/SOP/Entities/Computer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SOP.Entities
{
    public class Computer
    {
        [Key]
        [ForeignKey("Item")]
        public int Id { get; set; }

        public Item Item { get; set; }

        public List<Computer_ComputerPart> Computer_ComputerParts { get; set; }
    }
}
